src/components/Header/components/Login.tsx

```diff
@@ -1,61 +1,97 @@
 import React from "react";
-import { chakra } from "@chakra-ui/react";
+import { Box, chakra, VStack } from "@chakra-ui/react";
 import FocusLock from "@chakra-ui/focus-lock";
 
 import Button from "../../Button/Button";
 import Icon from "../../Icons/Icon";
 import Link from "../../Link/Link";
 import List from "../../List/List";
+import Text from "../../Text/Text";
 import { LoginProps } from "./UpperNav";
 
-const Login = chakra(({ loginOpen, setLoginOpen }: LoginProps) => {
+const Login = chakra(({ loginOpen, patronName, setLoginOpen }: LoginProps) => {
   return (
     <FocusLock isDisabled={!loginOpen}>
       <Button
         buttonType="link"
-        className={`withIcon ${loginOpen ? "closeLoginButton" : "loginButton"}`}
+        className={`withIcon ${
+          loginOpen
+            ? "closeLoginButton"
+            : patronName
+            ? "accountButton"
+            : "loginButton"
+        }`}
         id="loginButton"
         onClick={() => setLoginOpen(!loginOpen)}
       >
-        {loginOpen ? "Close" : "Log In"}
+        {loginOpen ? "Close" : patronName ? "My Account" : "Log In"}
         <Icon align="right" name={loginOpen ? "close" : "arrow"} size="small" />
       </Button>
       {loginOpen && (
-        <List
-          id="login"
-          className="loginMenu"
-          noStyling
-          type="ul"
-          listItems={[
-            <Link id="logInCatalog" key="logInCatalog" href="#" type="button">
-              <Icon
-                align="left"
-                color="ui.white"
-                id="log-in-catalog-icon"
-                name="utilityAccountFilled"
-                size="medium"
-                title="Log in to your account"
-              />
-              Log Into The Catalog
-            </Link>,
+        <VStack className="loginMenu">
+          {patronName && (
+            <Box id="patronGreeting">
```

Comment:
In the current header, when this menu is opened, this text is focused to automatically. It should be replicated here. You can add a `ref` (and maybe a tabIndex value?) to this `Box` and through a `useEffect`, focus on this element when `loginOpen` is true.